<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Compositional Data Analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/paper.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 64px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 69px;
  margin-top: -69px;
}

.section h2 {
  padding-top: 69px;
  margin-top: -69px;
}
.section h3 {
  padding-top: 69px;
  margin-top: -69px;
}
.section h4 {
  padding-top: 69px;
  margin-top: -69px;
}
.section h5 {
  padding-top: 69px;
  margin-top: -69px;
}
.section h6 {
  padding-top: 69px;
  margin-top: -69px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Ro Allen</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-user"></span>
     
     Home
  </a>
</li>
<li>
  <a href="blog.html">
    <span class="fa fa-coffee"></span>
     
     Blog
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-spinner"></span>
     
     Lab Resources
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="bioconductor.html"> Bioconductor Pipeline</a>
    </li>
    <li class="dropdown-header"> Exploratory Analysis</li>
    <li class="dropdown-header"> R Chunks</li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="publications.html">
    <span class="fa fa-mortar-board"></span>
     
     Publications
  </a>
</li>
<li>
  <a href="cv.html">
    <span class="fa fa-file-text"></span>
     
     CV
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Compositional Data Analysis</h1>

</div>


<p style="text-align:justify">
Analysis of high-throughput sequencing data (e.g. 16S amplicon sequencing, RNA-seq, metagenomics, metatranscriptomics) is complicated by the fact that these data are invariably compositional. This is the case because sequencing technologies (e.g. <em>Illumina MiSeq</em>) return an arbitrary total number of reads per sample. Subsequently, these data only hold valuable information about the relative composition of samples, not absolute abundances. Compositional data do not lend themselves well to classical statistical analyses, as these data violate assumptions of independence (Fig. 1). Fortunately, a range of compositional analyses are currently available for handling high-throughput sequencing data. For a brief introduction to the challenge of compositionality in high-throughput sequencing data, see <a href="https://www.frontiersin.org/articles/10.3389/fmicb.2017.02224/full">Microbiome Datasets Are Compositional: And This Is Not Optional</a>.
</p>
<p>
 
</p>
<div class="figure">
<img src="coda_files/figure-html/unnamed-chunk-1-1.png" alt="&lt;font size=2&gt;&lt;b&gt;Figure 1.&lt;/b&gt; An illustration of the lack of independence between features in high-throughput sequencing data. &lt;b&gt;A:&lt;/b&gt; The true abundance of each ASV in each condition. &lt;b&gt;B:&lt;/b&gt; The proportion of each ASV in each condition. When dealing with compositional data, it appears that ASV 2-10 are less abundant in the treatment condition compared to the control condition. In reality the true abundances are identical. This occurs as the increased abundance of ASV 1 in the treatment condition incidentally reduces the proportion of reads attributed to ASV 2-10. &lt;b&gt;C &amp; D:&lt;/b&gt; This further example demonstrates that an identical proportion of features (e.g. B &amp; D) can be reached through entirely different paths, and that it is not possible to resolve the path from the compositional data alone.&lt;/font&gt;" width="1440" />
<p class="caption">
<font size=2><b>Figure 1.</b> An illustration of the lack of independence between features in high-throughput sequencing data. <b>A:</b> The true abundance of each ASV in each condition. <b>B:</b> The proportion of each ASV in each condition. When dealing with compositional data, it appears that ASV 2-10 are less abundant in the treatment condition compared to the control condition. In reality the true abundances are identical. This occurs as the increased abundance of ASV 1 in the treatment condition incidentally reduces the proportion of reads attributed to ASV 2-10. <b>C &amp; D:</b> This further example demonstrates that an identical proportion of features (e.g. B &amp; D) can be reached through entirely different paths, and that it is not possible to resolve the path from the compositional data alone.</font>
</p>
</div>
<p>
 
</p>
<p style="text-align:justify">
Below, I provide an introduction to compositional beta-diversity and differential abundance anaylses of high throughput sequencing data in the R environment. To test differences between communities, Aitchison’s distance is used as an input for PERMANOVA, executed in the <a href="https://cran.r-project.org/web/packages/vegan/index.html">Vegan</a> R package using the <em>adonis</em> function. Differential abundance of features (ASVs) are tested using the <a href="https://bioconductor.org/packages/release/bioc/html/ALDEx2.html">ALDEx2</a> method. The input data are structured as a phyloseq object. Note that the optimal analyses depend on the data in question.
</p>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
