---
title: "Compositional Data Analysis"
---
<p style="text-align:justify">
Analysis of high-throughput sequencing data (e.g. 16S amplicon sequencing, RNA-seq, metagenomics, metatranscriptomics) is complicated by the fact that these data are invariably compositional. This is the case because sequencing technologies (e.g. *Illumina MiSeq*) return an arbitrary total number of reads per sample. Subsequently, these data only hold valuable information about the relative composition of samples, not absolute abundances. Compositional data do not lend themselves well to classical statistical analyses, as these data violate assumptions of independence (Fig. 1). Fortunately, a range of compositional analyses are currently available for handling high-throughput sequencing data. For a brief introduction to the challenge of compositionality in high-throughput sequencing data, see [Microbiome Datasets Are Compositional: And This Is Not Optional](https://www.frontiersin.org/articles/10.3389/fmicb.2017.02224/full).  
</p>
<p>&nbsp;</p>

```{r, echo=FALSE, eval=TRUE, message=FALSE, fig.width=15, fig.height=10, fig.cap=paste("<font size=2><b>Figure 1.</b> An illustration of the lack of idenpendence between features in high-throughput sequencing data. <b>A:</b> The true abundance of each ASV in each condition. <b>B:</b> The proportion of each ASV in each condition. When dealing with proportional data, it appears that ASV 2-10 are less abundant in the treatment condition compared to the control condition. In reality the true abundances are identical. This occurs as the increased abundance of ASV 1 in the treatment condition incidentally reduces the proportion of reads attributed to ASV 2-10.</font>")}
library(ggplot2)
library(cowplot)

rx = matrix(nrow = 10, ncol = 2)
rx[,1] = 10
rx[,2] = 10
rx[1,2] = 20
rd = as.data.frame(rx)
rd = lapply(rd, function(x) x/sum(x))
rv = cbind(rd$V1, rd$V2)
rv = as.vector(as.matrix(rv))
rvd = as.data.frame(rv)
rvd$Condition = c(rep("Control",10), rep("Treatment", 10))
rvd$species = c("ASV 1", "ASV 2", "ASV 3", "ASV 4", "ASV 5", "ASV 6", "ASV 7", "ASV 8", "ASV 9", "ASV 10", "ASV 1", "ASV 2", "ASV 3", "ASV 4", "ASV 5", "ASV 6", "ASV 7", "ASV 8", "ASV 9", "ASV 10")
rvd$species = factor(rvd$species, levels = c("ASV 1", "ASV 2", "ASV 3", "ASV 4", "ASV 5", "ASV 6", "ASV 7", "ASV 8", "ASV 9", "ASV 10"))

ra = matrix(nrow = 10, ncol = 2)
ra[,1] = 10
ra[,2] = 10
ra[1,2] = 20
ra = as.data.frame(ra)
rva = cbind(ra[1], ra[2])
rva = as.vector(as.matrix(rva))
rva = as.data.frame(rva)
rva$Condition = c(rep("Control",10), rep("Treatment", 10))
rva$species = c("ASV 1", "ASV 2", "ASV 3", "ASV 4", "ASV 5", "ASV 6", "ASV 7", "ASV 8", "ASV 9", "ASV 10", "ASV 1", "ASV 2", "ASV 3", "ASV 4", "ASV 5", "ASV 6", "ASV 7", "ASV 8", "ASV 9", "ASV 10")
rva$species = factor(rva$species, levels = c("ASV 1", "ASV 2", "ASV 3", "ASV 4", "ASV 5", "ASV 6", "ASV 7", "ASV 8", "ASV 9", "ASV 10"))
#55 tot sum, 10 = 18.1818, ASV 2-10 = 5 (n=9, tot = 45)
rq = matrix(nrow = 10, ncol = 2)
rq[,1] = 10
rq[,2] = 5
rq[1,2] = 10
rq = as.data.frame(rq)
rvq = lapply(rq, function(x) x/sum(x))
rvq = cbind(rvq$V1, rvq$V2)
rvq = as.vector(as.matrix(rvq))
rvq = as.data.frame(rvq)
rvq$Condition = c(rep("Control",10), rep("Treatment", 10))
rvq$species = c("ASV 1", "ASV 2", "ASV 3", "ASV 4", "ASV 5", "ASV 6", "ASV 7", "ASV 8", "ASV 9", "ASV 10", "ASV 1", "ASV 2", "ASV 3", "ASV 4", "ASV 5", "ASV 6", "ASV 7", "ASV 8", "ASV 9", "ASV 10")
rvq$species = factor(rvd$species, levels = c("ASV 1", "ASV 2", "ASV 3", "ASV 4", "ASV 5", "ASV 6", "ASV 7", "ASV 8", "ASV 9", "ASV 10"))

rw = matrix(nrow = 10, ncol = 2)
rw[,1] = 10
rw[,2] = 5
rw[1,2] = 10
rw = as.data.frame(rw)
rvw = cbind(rw[1], rw[2])
rvw = as.vector(as.matrix(rvw))
rvw = as.data.frame(rvw)
rvw$Condition = c(rep("Control",10), rep("Treatment", 10))
rvw$species = c("ASV 1", "ASV 2", "ASV 3", "ASV 4", "ASV 5", "ASV 6", "ASV 7", "ASV 8", "ASV 9", "ASV 10", "ASV 1", "ASV 2", "ASV 3", "ASV 4", "ASV 5", "ASV 6", "ASV 7", "ASV 8", "ASV 9", "ASV 10")
rvw$species = factor(rvw$species, levels = c("ASV 1", "ASV 2", "ASV 3", "ASV 4", "ASV 5", "ASV 6", "ASV 7", "ASV 8", "ASV 9", "ASV 10"))

#end
p1 = ggplot(rvd, aes(species, rv, fill = Condition)) + geom_bar(stat = "identity", position = "dodge") + theme_bw() + ylab("Proportion") + xlab("Amplicon Sequence Variant") + 
  scale_y_continuous(expand = c(0, 0), limits = c(0, 0.21)) + 
  scale_fill_manual(values = c("skyblue2", "darkblue"))

p2 = ggplot(rva, aes(species, rva, fill = Condition)) + geom_bar(stat = "identity", position = "dodge") + theme_bw() + ylab("Abundance") + xlab("Amplicon Sequence Variant") + 
  scale_y_continuous(expand = c(0, 0), limits = c(0, 21)) + 
  scale_fill_manual(values = c("skyblue2", "darkblue"))

p3 = ggplot(rvq, aes(species, rvq, fill = Condition)) + geom_bar(stat = "identity", position = "dodge") + theme_bw() + ylab("Proportion") + xlab("Amplicon Sequence Variant") + 
  scale_y_continuous(expand = c(0, 0), limits = c(0, 0.21)) + 
  scale_fill_manual(values = c("lightcoral", "darkred"))

p4 = ggplot(rvw, aes(species, rvw, fill = Condition)) + geom_bar(stat = "identity", position = "dodge") + theme_bw() + ylab("Abundance") + xlab("Amplicon Sequence Variant") + 
  scale_y_continuous(expand = c(0, 0), limits = c(0, 21)) + 
  scale_fill_manual(values = c("lightcoral", "darkred"))

plot_grid(p2, p1, p4, p3, labels = c('A', 'B', 'C', 'D'), ncol = 2)
```
<p>&nbsp;</p>
<p style="text-align:justify">
Below, I provide an introduction to compositional beta-diversity and differential abundance anaylses of high throughput sequencing data in the R environment. To test differences between communities, Aitchison's distance is used as an input for PERMANOVA, executed in the [Vegan](https://cran.r-project.org/web/packages/vegan/index.html) R package using the *adonis* function. Differential abundance of features (ASVs) are tested using the [ALDEx2](https://bioconductor.org/packages/release/bioc/html/ALDEx2.html) method. The input data are structured as a phyloseq object. Note that the optimal analyses depend on the data in question. 
</p>
