<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>External Resources</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/paper.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 64px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 69px;
  margin-top: -69px;
}

.section h2 {
  padding-top: 69px;
  margin-top: -69px;
}
.section h3 {
  padding-top: 69px;
  margin-top: -69px;
}
.section h4 {
  padding-top: 69px;
  margin-top: -69px;
}
.section h5 {
  padding-top: 69px;
  margin-top: -69px;
}
.section h6 {
  padding-top: 69px;
  margin-top: -69px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Ro Allen</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-user"></span>
     
     Home
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-spinner"></span>
     
     Lab Resources
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="bioconductor.html"> Bioconductor Pipeline</a>
    </li>
    <li class="dropdown-header"> Exploratory Analysis</li>
    <li>
      <a href="oceanmap.html"> Oceanmap Tutorial</a>
    </li>
    <li>
      <a href="rchunks.html"> R Chunks</a>
    </li>
    <li>
      <a href="exres.html"> External Resources</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="publications.html">
    <span class="fa fa-mortar-board"></span>
     
     Publications
  </a>
</li>
<li>
  <a href="cv.html">
    <span class="fa fa-file-text"></span>
     
     CV
  </a>
</li>
<li>
  <a href="https://twitter.com/rojallen">
    <span class="fa fa-twitter"></span>
     
    Twitter
  </a>
</li>
<li>
  <a href="https://github.com/ro-allen">
    <span class="fa fa-github"></span>
     
    Github
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">External Resources</h1>

</div>


<p style="text-align:justify">
<p>Below is a collection of useful external resources for data processing and analysis in microbial ecology.</p>
<p><a href="http://web.stanford.edu/class/bios221/book/">Modern Statistics for Modern Biology</a> An easy to digest statistics textbook with a focus on high-throughput sequencing data, written by Prof. Susan Holmes and Dr. Wolfgang Huber.</p>
<p><a href="https://dx.doi.org/10.12688/f1000research.8986.2">Bioconductor workflow for microbiome data analysis: from raw reads to community analyses</a> F1000 publication (Callahan et al. 2016) of the bioconductor pipeline for microbiome data analysis. The paper includes R scripts which are clearly annotated.</p>
<p><a href="https://mb3is.megx.net/gustame">GUSTA ME: Guide to statistical analysis in microbial ecology</a> A live resource which acts as an excellent primer for common analyses in microbial ecology, with particularly clear and concise explanations of different ordination techinques.</p>
<p><a href="http://r4ds.had.co.nz">R for Data Science</a> A key text for R in data science, authored by Garrett Grolemund and Hadley Wickham. This text is free to read online, and covers essentials for using R for data analysis.</p>
<p><a href="http://nickstrayer.me/visualization_in_r/">Best Practives in Dataviz: An R Perspective</a> Interesting commentary on how the choice of visualisation method can influence the interpretation of data. Worth reading when deciding how best to represent your findings.</p>
<p><a href="https://ourcodingclub.github.io">Coding Club</a> A community based at Edinburgh University focussed on peer-learning statistical skills in R. The site is worth reading regularly, and is an great resource when approaching new challenges with R and its common packages.</p>
<p><a href="https://www.youtube.com/channel/UC_R5smHVXRYGhZYDJsnXTwg/videos">R Consortium</a> The R consortium <em>YouTube</em> channel is full of insightful videos from R package developers, including talks from useR conferences. These act as a good resource from broadening your knowledge of R.</p>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
