<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Home</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="cyborg_mod.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Ro Allen</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-user"></span>
     
     Home
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-spinner"></span>
     
     Lab Resources
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="bioconductor.html"> Bioconductor Pipeline</a>
    </li>
    <li>
      <a href="oceanmap.html"> Oceanmap Tutorial</a>
    </li>
    <li>
      <a href="rchunks.html"> R Chunks</a>
    </li>
    <li>
      <a href="exres.html"> External Resources</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="publications.html">
    <span class="fa fa-mortar-board"></span>
     
     Publications
  </a>
</li>
<li>
  <a href="cv.html">
    <span class="fa fa-file-text"></span>
     
     CV
  </a>
</li>
<li>
  <a href="https://twitter.com/rojallen">
    <span class="fa fa-twitter"></span>
     
    Twitter
  </a>
</li>
<li>
  <a href="https://github.com/ro-allen">
    <span class="fa fa-github"></span>
     
    Github
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">




</div>


<div>
<p><img src="clayton-tonna-17EfL8ACKvQ-unsplash.jpg" /></p>
</div>
<div class="mycontent">
<div class="col-sm-8">
<blockquote>
<h3>
Research Statement
</h3>
<p style="text-align:justify">
My goal is to advance our fundamental understanding of marine microbial ecology, leveraging rapidly developing molecular and statistical techniques to approach hypothesis-driven studies. This understanding, ultimately, will contribute to our ability to predict, manipulate, and harness microbial communities, with far-reaching implications across a range of systems.
</p>
<p>
<a href="cv.html" class="btn btn-primary">View CV</a>
</p>
</blockquote>
</div>
<div class="col-sm-4">
<a class="twitter-timeline"
   href="https://twitter.com/rojallen"
   data-width="300"
   data-height="300"
   data-chrome="nofooter noscrollbar noborders transparent"
   data-theme="dark"
   data-tweet-limit="1">Tweets by <span class="citation">@rojallen</span></a>
<script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
</div>
</div>
</div>
<p>
 
</p>
<p>
 
</p>
<div class="mycontent">
<p style="text-align:justify">
I am currently a Postdoc in <a href="https://www.mba.ac.uk/fellows/cunliffe-group">Dr. Michael Cunliffe’s</a> laboratory at the Marine Biological Association of the United Kindgom, studying planktonic marine fungi. I previously completed my PhD at the University of Otago Department of Botany, supervised by <a href="https://www.researchgate.net/profile/Linn_Hoffmann">Dr. Linn Hoffmann</a> and <a href="https://www.researchgate.net/profile/Tina_Summerfield">Dr. Tina Summerfield</a>, and an M.Res. in marine biology at Plymouth University and the Marine Biological Association supervised by <a href="https://www.researchgate.net/profile/Jason_Hall-Spencer">Prof. Jason Hall-Spencer</a> and <a href="https://www.researchgate.net/profile/Andrew_Foggo">Dr. Andy Foggo</a>. My research interests centre around the fundamental ecological dynamics of the microbial communities, whilst my PhD thesis focussed on the use of high-throughput amplicon sequencing to investigate the community level responses of marine protists and bacteria to climate change.
</p>
<p>
 
</p>
<h4 id="current-projects">Current Projects</h4>
<hr />
<p><b>Microbial community assembly in the oligotrophic ocean</b></p>
<p style="text-align:justify">
The patterns and processes involved in microbial community assembly in the marine environment are not well understood, particularly in the oligotrophic open ocean where data on the structure and distribution of microbial communites are scarce. During the ~2500 km <em>GEOTRACES</em> transect of the South Pacific Gyre aboard the <a href="https://www.niwa.co.nz/services/vessels/niwa-vessels/rv-tangaroa">RV Tangaroa</a>, we sampled seawater from 24 locations at 4 depths to determine microbial community structure (16S amplicon sequencing) alongiside a suite of environmental parameters in order to investigate the drivers of microbial community assembly in these regions. These high-resolution data will act to inform fundamental mechanisms underpinning the distribution of microbes in the oligotrophic ocean, and have implications for the region in a changing environment.
</p>
<p>
 
</p>
<p><b>Bacterioplankton responses to climate change in the oligotrophic ocean</b></p>
<p style="text-align:justify">
Oligotrophic oceans represent ~40% of total ocean surface area, and account ~20% of marine primary production. Bacterioplankton are responsible for the majority of photosynthesis, nitrogen fixation, and recycling of organic material in these regions. The productivity of the oligotrophic ocean is closely coupled with the efficiency of organic matter recycling by heterotrophic bacteria as this process largely controls nutrient availability, due to limited inputs from vertical transport and low rates of nitrogen fixation. Subsequently, the responses of oligotrophic ocean bacterioplankton communities to ocean acidification and warming are likely to influence overall marine productivity and the global carbon cycle. During the <em>GEOTRACES</em> transect through the oligotrophic ocean aboard the <a href="https://www.niwa.co.nz/services/vessels/niwa-vessels/rv-tangaroa">RV Tangaroa</a>, a team of scientists from NIWA and the University of Otago conducted a series of trace-metal clean deckboard incubation experiments to generate early data on this topic.
</p>
<p>
 
</p>
<p><b>Coastal eukaryotic phytoplankton responses to ocean acidification and warming</b></p>
<p style="text-align:justify">
The phytoplankton are a phylogenetically diverse group of photoautotrophic organisms, grouped together as a result of their functional similarity and shared occupancy of the photic ocean. Approximately 30% of global primary production can be attributed to eukaryotic phytoplankton, and these organisms underpin marine foodwebs. As part of New Zealand’s interdisciplinary large-scale <a href="http://www.carim.nz">CARIM</a> mesocosm project, we aim characterise eukaryotic phytoplankton community responses to ocean acidification and warming and identify trends in genetic diversity within specific taxonomic groups using 18S rRNA gene amplicon sequencing and qPCR.
</p>
<p>
 
</p>
<p><b>The impact of ocean acidification on marine biofilms dynamics: a CO<sub>2</sub> seep study</b></p>
<p style="text-align:justify">
Marine biolfilms are heterogenous microhabitats dominated by protists and bacteria. Marine biofilms provide a fundamental grazing resource for larger organisms such as gastropods in shallow marine ecosystems, whilst bacteria in biofilms play a further key role by cuing settlement in scleractinian corals and other habitat forming organisms. Understanding the effects of ocean acidification on biofilm community composition and succession from both a eukaryotic and prokaryotic perspective is critical, as potential shifts will have ramifications for the settlement of habitat forming organisms, trophic energy transfer in shallow marine ecosystems, and biodiversity directly associated with biofilms. High-throughput amplicon sequencing has opened the door to high-resolution profiling of marine biofilms. Here, we aim to characterise the effect of ocean acidification on biofilm community composition and succession by using volcanic CO<sub>2</sub> seeps as a proxy of future acidified conditions.
</p>
<p>
 
</p>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
