<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title></title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/paper.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 64px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 69px;
  margin-top: -69px;
}

.section h2 {
  padding-top: 69px;
  margin-top: -69px;
}
.section h3 {
  padding-top: 69px;
  margin-top: -69px;
}
.section h4 {
  padding-top: 69px;
  margin-top: -69px;
}
.section h5 {
  padding-top: 69px;
  margin-top: -69px;
}
.section h6 {
  padding-top: 69px;
  margin-top: -69px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Ro Allen</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-user"></span>
     
     Home
  </a>
</li>
<li>
  <a href="blog.html">
    <span class="fa fa-coffee"></span>
     
     Blog
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-spinner"></span>
     
     Lab Resources
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="bioconductor.html"> Bioconductor Pipeline</a>
    </li>
    <li class="dropdown-header"> Exploratory Analysis</li>
    <li class="dropdown-header"> R Chunks</li>
    <li class="dropdown-header"> Protocols</li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="publications.html">
    <span class="fa fa-mortar-board"></span>
     
     Publications
  </a>
</li>
<li>
  <a href="cv.html">
    <span class="fa fa-file-text"></span>
     
     CV
  </a>
</li>
<li>
  <a href="https://twitter.com/rojallen">
    <span class="fa fa-twitter"></span>
     
    Twitter
  </a>
</li>
<li>
  <a href="https://github.com/ro-allen">
    <span class="fa fa-github"></span>
     
    Github
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">




</div>


<p>
 
</p>
<p>
 
</p>
<center>
<div style="width:300px; height=200px">
<div class="figure">
<img src="Github_pic.png" />

</div>
</div>
</center>
<p>
 
</p>
<p style="text-align:justify">
I am a PhD candidate at the University of Otago Department of Botany, supervised by <a href="https://www.researchgate.net/profile/Linn_Hoffmann">Dr. Linn Hoffmann</a> and <a href="https://www.researchgate.net/profile/Tina_Summerfield">Dr. Tina Summerfield</a>. Previously, I completed an M.Res. in marine biology at Plymouth University supervised by <a href="https://www.researchgate.net/profile/Jason_Hall-Spencer">Prof. Jason Hall-Spencer</a> and <a href="https://www.researchgate.net/profile/Andrew_Foggo">Dr. Andy Foggo</a>. My research interests centre around the fundamental dynamics of the marine microbiome, whilst my PhD thesis focusses on the use of high-throughput amplicon sequencing to investigate the community level responses of marine protists and bacteria to climate change. I am particularly interested in the handling of high-throughput sequencing data.
</p>
<p>
 
</p>
<div id="current-projects" class="section level3">
<h3>Current Projects</h3>
<hr />
<p><b>Microbial community assembly in the oligotrophic ocean</b></p>
<p style="text-align:justify">
The patterns and processes involved in microbial community assembly in the marine environment are not well understood, particularly in the oligotrophic open ocean where data on the structure and distribution of microbial communites are scarce. During the ~2000 km <em>GEOTRACES</em> transect of the South Pacific Gyre aboard the <a href="https://www.niwa.co.nz/services/vessels/niwa-vessels/rv-tangaroa">RV Tangaroa</a>, we sampled seawater from 24 locations at 4 depths to determine microbial community structure (16S amplicon sequencing) alongiside a suite of environmental parameters in order to investigate the drivers of microbial community assembly in these regions. These high-resolution data will act to inform fundamental mechanisms underpinning the distribution of microbes in the oligotrophic ocean, and have implications for the region in a changing environment.
</p>
<p>
 
</p>
<p><b>Bacterioplankton responses to climate change in the oligotrophic ocean</b></p>
<p style="text-align:justify">
Oligotrophic oceans represent ~40% of total ocean surface area, and account ~20% of marine primary production. Bacterioplankton are responsible for the majority of photosynthesis, nitrogen fixation, and recycling of organic material in these regions. The productivity of the oligotrophic ocean is closely coupled with the efficiency of organic matter recycling by heterotrophic bacteria as this process largely controls nutrient availability, due to limited inputs from vertical transport and low rates of nitrogen fixation. Subsequently, the responses of oligotrophic ocean bacterioplankton communities to ocean acidification and warming are likely to influence overall marine productivity and the global carbon cycle. During the <em>GEOTRACES</em> transect through the oligotrophic ocean aboard the <a href="https://www.niwa.co.nz/services/vessels/niwa-vessels/rv-tangaroa">RV Tangaroa</a>, a team of scientists from NIWA and the University of Otago conducted a series of trace-metal clean deckboard incubation experiments to generate early data on this topic.
</p>
<p>
 
</p>
<p><b>Eukaryotic phytoplankton responses to ocean acidification and warming in New Zealand coastal waters</b></p>
<p style="text-align:justify">
The phytoplankton are a phylogenetically diverse group of autotrophic organisms, grouped together as a result of their functional similarity and shared occupancy of the photic ocean. Approximately 30% of global primary production can be attributed to eukaryotic phytoplankton, and these organisms underpin marine foodwebs. As part of New Zealand’s interdisciplinary <a href="http://www.carim.nz">CARIM</a> mesocosm project, we aim characterise eukaryotic phytoplankton community responses to ocean acidification and warming and identify trends in genetic diversity within specific taxonomic groups using 18S rRNA gene amplicon sequencing and qPCR.
</p>
<p>
 
</p>
<p><b>The impact of ocean acidification on marine biofilms: a CO<sub>2</sub> seep study</b></p>
<p style="text-align:justify">
Marine biolfilms are heterogenous microhabitats dominated by protists and bacteria. Marine biofilms provide a fundamental grazing resource for larger organisms such as gastropods in shallow marine ecosystems, whilst bacteria in biofilms play a further key role by cuing settlement in scleractinian corals and other habitat forming organisms. Understanding the effects of ocean acidification on biofilm community composition and succession from both a eukaryotic and prokaryotic perspective is critical, as potential shifts will have ramifications for the settlement of habitat forming organisms, trophic energy transfer in shallow marine ecosystems, and biodiversity directly associated with biofilms. High-throughput amplicon sequencing has opened the door to high-resolution profiling of marine biofilms. Here, we aim to characterise the effect of ocean acidification on biofilm community composition and succession by using volcanic CO<sub>2</sub> seeps as a proxy of future acidified conditions.
</p>
<p>
 
</p>
<p><b>Oceanic fronts: a tool for investigating phytoplankton-bacteria interactions</b></p>
<p style="text-align:justify">
Oceanic fronts are dynamic regions where distinct water masses converge. These regions provide an opportunity to investigate environmental modulation of phytoplankton-bacteria interactions. Using high spatial resolution sampling of the <a href="https://www.st.nmfs.noaa.gov/copepod/time-series/nz-10101/">Munida transect</a>, we aim to construct a region specific phytoplankton-bacteria interaction network and elucidate the influence of environmental conditions on these interactions.
</p>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
