<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>ISME 17 Supplementary Information</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/paper.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 64px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 69px;
  margin-top: -69px;
}

.section h2 {
  padding-top: 69px;
  margin-top: -69px;
}
.section h3 {
  padding-top: 69px;
  margin-top: -69px;
}
.section h4 {
  padding-top: 69px;
  margin-top: -69px;
}
.section h5 {
  padding-top: 69px;
  margin-top: -69px;
}
.section h6 {
  padding-top: 69px;
  margin-top: -69px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Ro Allen</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-user"></span>
     
     Home
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-spinner"></span>
     
     Lab Resources
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="bioconductor.html"> Bioconductor Pipeline</a>
    </li>
    <li class="dropdown-header"> Exploratory Analysis</li>
    <li>
      <a href="oceanmap.html"> Oceanmap Tutorial</a>
    </li>
    <li>
      <a href="rchunks.html"> R Chunks</a>
    </li>
    <li>
      <a href="exres.html"> External Resources</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="publications.html">
    <span class="fa fa-mortar-board"></span>
     
     Publications
  </a>
</li>
<li>
  <a href="cv.html">
    <span class="fa fa-file-text"></span>
     
     CV
  </a>
</li>
<li>
  <a href="https://twitter.com/rojallen">
    <span class="fa fa-twitter"></span>
     
    Twitter
  </a>
</li>
<li>
  <a href="https://github.com/ro-allen">
    <span class="fa fa-github"></span>
     
    Github
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">ISME 17 Supplementary Information</h1>

</div>


<p style="text-align:justify">
Microbial community assembly processes in the South Pacific Gyre were resolved using a null-model approach described in <a href="https://www.nature.com/articles/ismej201393">Stegen et al. (2013)</a>. A fundamental assumption of the approach is that microbes which are more phylogenetically similar also occupy more similar ecological niches. This assumption can be empirically tested first by generating two distance matrices. The first, of phylogenetic similarity between each pair of amplicon sequence variants (ASVs) based on 16S rRNA gene sequences, and the second, of ecological niche similarity between each pair of ASVs based on weighted occurence values. These two distance matrices are then used to generate a mantel correlogram (Fig. 1) which illustrates the correlation between the two matrices at varying levels of phylogenetic similarity.
</p>
<p style="text-align:justify">
Figure 1 shows that significant positive correlation between phylogenetic similarity and ecological niche similarity is constrained to only very closely related ASVs. This is accounted for by using the <span class="math inline">\(\beta\)</span>-mean nearest taxon distance meaure to quantify phylogenetic turnover between communities.
</p>
<center>
<div style="width:400px; height=266px">
<div class="figure">
<img src="Correlogram.jpeg" />

</div>
</div>
</center>
<p style="text-align:justify">
<strong>Fig. 1</strong> Mantel correlogram of the relationship between phylogenetic distance and ecological niche distance in microbial communities in the South Pacific Gyre. Ecological niche was calculated taking into account all measured environmental variables.
</p>
<p>
Â 
</p>
<p style="text-align:justify">
Below, I briefly introduce the measures reported in the poster, and the overall workflow used to resolve microbial communtiy assembly processes:
</p>
<p style="text-align:justify">
<strong><span class="math inline">\(\beta\)</span>-mean nearest taxon distance (<span class="math inline">\(\beta\)</span>MNTD)</strong> is a measure of phylogenetic turnover between communities. The metric quantifies the phylogenetic distance between each ASV in community A, and its closest relative in community B, generating an abundance-weighted mean value for each pair of communities.
</p>
<p style="text-align:justify">
<strong><span class="math inline">\(\beta\)</span>-nearest taxon index (<span class="math inline">\(\beta\)</span>NTI)</strong> is a measure of the relationship between observed <span class="math inline">\(\beta\)</span>MNTD and the distribution of <span class="math inline">\(\beta\)</span>MNTD values expected under a null model distribution (simulating neutral community assembly). The measure is expressed as standard deviation according to the null model distribution. <span class="math inline">\(\beta\)</span>MNTD values &gt; +2 show greater phylogenetic turnover than expected, indicating that <em>variable selection</em> is the dominant microbial community assembly process. In contrast, values &lt; -2 show less phylogenetic turnover than expected, indicating that <em>homogenising selection</em> is the dominant microbial community assembly process. pairs of communities with non significant <span class="math inline">\(\beta\)</span>NTI values (&gt; |2|) are further analyses using the RC<sub>bray</sub> measure described below.
</p>
<p style="text-align:justify">
<strong>Raup-Crick<sub>bray</sub> (RC<sub>bray</sub>)</strong> is a measure of the relationship between observed bray-curtis dissimilarity and the distribution of bray-curtis dissimilarity values expected under a null model. The metric is superificially similar to <span class="math inline">\(\beta\)</span>NTI, but does not take phylogenetic information into account, and can subsequently be used to resolve stochastic community assembly processes. RC<sub>bray</sub> values &gt; +0.95 show that there is greater ASV turnover between communities than expected under the null model, indicating that <em>dispersal limitation</em> is the dominant community assembly process. RC<sub>bray</sub> values &lt; =0.95 show that there is less ASV turnover between communities than expected under the null model, indicating that <em>homogenising dispersal</em> is the dominant community assembly process. RC<sub>bray</sub> values &lt; |0.95| represent the undominated community assembly scenario described in <a href="https://www.frontiersin.org/articles/10.3389/fmicb.2015.00370/full">Stegen et al. (2015)</a>.
</p>
<p style="text-align:justify">
For further information of the measures described above, see <a href="https://www.nature.com/articles/ismej201393">Stegen et al. (2013)</a>.
</p>
<p style="text-align:justify">
<p>The analyses are implemented according to the workflow shown in Figure 2, where observed and null model distribution <span class="math inline">\(\beta\)</span>MNTD values are calculated for each pair of communities. From these values, <span class="math inline">\(\beta\)</span>NTI is calculated. If pairs communities display non-significant <span class="math inline">\(\beta\)</span>NTI values, they are subjected to RC<sub>bray</sub> analysis to partition stochastic community assembly processes.</p>
<center>
<div style="width:400px; height=266px">
<div class="figure">
<img src="isme_model.png" />

</div>
</div>
</center>
<p style="text-align:justify">
<strong>Fig. 2</strong> From <a href="https://www.nature.com/articles/ismej201393">Stegen et al. (2013)</a>. Diagrammatic summary of the workflow used to resolve microbial community assembly processes.
</p>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
